src/dialogs/more-info/ha-more-info-dialog.ts

```diff
@@ -0,0 +1,298 @@
+import { navigate } from "../../common/navigate";
+import "../../components/state-history-charts";
+import { removeEntityRegistryEntry } from "../../data/entity_registry";
+import { showEntityEditorDialog } from "../../panels/config/entities/show-dialog-entity-editor";
+import "../../state-summary/state-card-content";
+import { showConfirmationDialog } from "../generic/show-dialog-box";
+import "./more-info-content";
+import { customElement, LitElement, property, css } from "lit-element";
+import { html } from "lit-html";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { fireEvent } from "../../common/dom/fire_event";
+import { getRecentWithCache, CacheConfig } from "../../data/cached-history";
+import { computeDomain } from "../../common/entity/compute_domain";
+import { mdiClose, mdiSettings, mdiPencil } from "@mdi/js";
+
+const DOMAINS_NO_INFO = ["camera", "configurator", "history_graph"];
+const EDITABLE_DOMAINS_WITH_ID = ["scene", "automation"];
+const EDITABLE_DOMAINS = ["script"];
+
+@customElement("ha-more-info-dialog")
+export class MoreInfoDialog extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() private _stateHistory?: object;
+
+  @property({ type: Boolean, reflect: true }) public large = false;
+
+  @property() private _cacheConfig?: CacheConfig;
+
+  private _interval?: number;
+
+  protected updated(changedProperties) {
+    super.updated(changedProperties);
+    if (changedProperties.has("hass")) {
+      const oldHass = changedProperties.get("hass");
+      if (!oldHass || oldHass.moreInfoEntityId !== this.hass.moreInfoEntityId) {
+        if (this.hass.moreInfoEntityId) {
+          this.setAttribute(
+            "data-domain",
+            computeDomain(this.hass.moreInfoEntityId)
+          );
+          this.large = false;
+          this._stateHistory = undefined;
+          if (this._computeShowHistoryComponent(this.hass.moreInfoEntityId)) {
+            this._cacheConfig = {
+              refresh: 60,
+              cacheKey: `more_info.${this.hass.moreInfoEntityId}`,
+              hoursToShow: 24,
+            };
+            this._getStateHistory();
+            clearInterval(this._interval);
+            if (this._cacheConfig.refresh) {
+              this._interval = window.setInterval(() => {
+                this._getStateHistory();
+              }, this._cacheConfig.refresh * 1000);
+            }
+          }
+        } else {
+          this._stateHistory = undefined;
+          clearInterval(this._interval);
+          this._interval = undefined;
+        }
+      }
+    }
+  }
+
+  protected render() {
+    if (!this.hass.moreInfoEntityId) {
+      return html``;
+    }
+    const entityId = this.hass.moreInfoEntityId;
+    const stateObj = this.hass.states[entityId];
+    const domain = computeDomain(entityId);
+
+    return html`
+      <ha-dialog open @closed=${this._close} .heading=${true} hideActions>
```

Comment:
Can we make `heading` a boolean attribute so we can just make this 
```suggestion
      <ha-dialog open @closed=${this._close} heading hideActions>
```

I think we might also want to rename it to something that better describes what it does. `renderHeading` ?

Reply:
it is a hack, it is from `mwc-dialog` it should be a string they only call `renderHeading` when `heading` is truthy, but we want to use the slot...

Reply:
lol ok